lib/computer_player.rb

```diff
@@ -1,17 +1,80 @@
+require_relative 'win_checker'
+require_relative 'score_tracker'
+
 class ComputerPlayer < Player
 
-  def initialize(player_num, board)
-    @board = board
+  def initialize(player_num, side_length)
+    @player_num = player_num
+    @next_move = 0
+    @side_length = side_length
+    @comp_token = "O"
+    @other_token = "X"
   end
 
-  def get_next_move
-    available_moves = @board.get_available_moves
+  def get_next_move(board)
+    minimax_score(board.to_string_array, @player_num)
 
-    available_moves[0].to_s
+    @next_move.to_s
   end
 
   def get_token
-    "O"
+    @comp_token
+  end
+
+  private
+
+  def minimax_score(board_string_array, next_player)
+    current_board = Board.new(@side_length)
+    current_board.set_board(board_string_array)
+    score_tracker = ScoreTracker.new(current_board)
+
+    player_that_just_went = (next_player == 1) ? 2 : 1
+
+    player_that_just_went_token = get_player_token(player_that_just_went) 
+    score_tracker.update_scores(player_that_just_went_token,
+                                current_board.number_of_moves_left)
+
+    current_score = score_tracker.get_player_score(@comp_token)
+
+    if score_tracker.is_game_finished?
+      return current_score
+    else
+      possible_moves = current_board.get_available_moves
+      
+      board_array = Array.new
+      possible_moves.length.times {
```

Comment:
Why do you do this loop twice (also line 54)?  Why store all these boards in an array before using them in the next loop?  The array doesn't seem to be used for anything.

Reply:
Only looping once now, and not using board_array anymore.